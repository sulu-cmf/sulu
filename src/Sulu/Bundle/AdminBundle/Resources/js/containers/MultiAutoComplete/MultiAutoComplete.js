// @flow
import React from 'react';
import {autorun} from 'mobx';
import type {IObservableValue} from 'mobx';
import {observer} from 'mobx-react';
import equals from 'fast-deep-equal';
import MultiAutoCompleteComponent from '../../components/MultiAutoComplete';
import SearchStore from '../../stores/SearchStore';
import MultiSelectionStore from '../../stores/MultiSelectionStore';

type Props = {|
    allowAdd: boolean,
    disabled: boolean,
    displayProperty: string,
    filterParameter: string,
    id?: string,
    idProperty: string,
    locale?: ?IObservableValue<string>,
    onChange: (value: Array<string | number>) => void,
    resourceKey: string,
    searchProperties: Array<string>,
    value: ?Array<string | number>,
|};

@observer
export default class MultiAutoComplete extends React.Component<Props> {
    static defaultProps = {
        allowAdd: false,
        disabled: false,
        filterParameter: 'ids',
        idProperty: 'id',
    };

    searchStore: SearchStore;
    selectionStore: MultiSelectionStore;
    changeDisposer: () => void;
    changeAutorunInitialized: boolean = false;

    constructor(props: Props) {
        super(props);

        const {
            filterParameter,
            idProperty,
            locale,
            onChange,
            resourceKey,
            searchProperties,
            value,
        } = this.props;

        this.searchStore = new SearchStore(resourceKey, searchProperties);
        this.selectionStore = new MultiSelectionStore(resourceKey, value || [], locale, filterParameter);

        this.changeDisposer = autorun(() => {
            const itemIds = this.selectionStore.items.map((item) => item[idProperty]);

            if (this.selectionStore.loading) {
                return;
            }

            if (!this.changeAutorunInitialized) {
                this.changeAutorunInitialized = true;
                return;
            }

            onChange(itemIds);
        });
    }

    componentWillUnmount() {
        this.changeDisposer();
    }

    componentDidUpdate(prevProps: Props) {
        const {value} = this.props;

        if (!equals(prevProps.value, value)) {
            this.selectionStore.loadItems(value);
        }
    }

    handleChange = (value: Array<Object>) => {
        this.selectionStore.set(value);
        this.searchStore.clearSearchResults();
    };

    handleSearch = (query: string) => {
        this.searchStore.search(query, this.selectionStore.items.map((item) => item.id));
    };

    render() {
        const {
            props: {
                allowAdd,
                disabled,
                displayProperty,
                id,
                idProperty,
                searchProperties,
            },
        } = this;

        return (
            <MultiAutoCompleteComponent
                allowAdd={allowAdd}
                disabled={disabled}
                displayProperty={displayProperty}
                id={id}
                idProperty={idProperty}
                loading={this.searchStore.loading || this.selectionStore.loading}
                onChange={this.handleChange}
                onSearch={this.handleSearch}
                searchProperties={searchProperties}
                suggestions={this.searchStore.searchResults}
                value={this.selectionStore.items || []}
            />
        );
    }
}
