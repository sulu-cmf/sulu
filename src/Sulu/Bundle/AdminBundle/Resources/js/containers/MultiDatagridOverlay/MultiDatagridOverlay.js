// @flow
import React from 'react';
import {observable } from 'mobx';
import type {IObservableValue} from 'mobx';
import DatagridStore from '../../containers/Datagrid/stores/DatagridStore';
import DatagridOverlay from '../DatagridOverlay';

type Props = {|
    adapter: string,
    allowActivateForDisabledItems?: boolean,
    clearSelectionOnClose: boolean,
    confirmLoading?: boolean,
    disabledIds?: Array<string | number>,
    locale?: ?IObservableValue<string>,
    onClose: () => void,
    onConfirm: (selectedItems: Array<Object>) => void,
    open: boolean,
    options?: Object,
    resourceKey: string,
    preSelectedItems: Array<Object>,
    title: string,
|};

export default class MultiDatagridOverlay extends React.Component<Props> {
    datagridStore: DatagridStore;
    page: IObservableValue<number> = observable.box(1);

    static defaultProps = {
        clearSelectionOnClose: false,
        preSelectedItems: [],
    };

    constructor(props: Props) {
        super(props);

        const {locale, options, preSelectedItems, resourceKey} = this.props;
        const observableOptions = {};
        observableOptions.page = this.page;

        if (locale) {
            observableOptions.locale = locale;
        }

        this.datagridStore = new DatagridStore(resourceKey, observableOptions, options);

        preSelectedItems.forEach((preSelectedItem) => {
            this.datagridStore.select(preSelectedItem);
        });
    }

    componentWillUnmount() {
        this.datagridStore.destroy();
    }

    handleConfirm = () => {
        this.props.onConfirm(this.datagridStore.selections);
    };

    render() {
        const {
            adapter,
            allowActivateForDisabledItems,
            clearSelectionOnClose,
            confirmLoading,
            disabledIds,
            onClose,
            open,
            preSelectedItems,
            title,
        } = this.props;

        return (
            <DatagridOverlay
                adapter={adapter}
                allowActivateForDisabledItems={allowActivateForDisabledItems}
                clearSelectionOnClose={clearSelectionOnClose}
                confirmLoading={confirmLoading}
                datagridStore={this.datagridStore}
                disabledIds={disabledIds}
                onClose={onClose}
                onConfirm={this.handleConfirm}
                open={open}
                preSelectedItems={preSelectedItems}
                title={title}
            />
        );
    }
}
