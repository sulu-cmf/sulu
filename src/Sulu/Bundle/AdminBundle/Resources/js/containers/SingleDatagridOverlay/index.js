// @flow
import SingleDatagridOverlay from './SingleDatagridOverlay';

export default SingleDatagridOverlay;
