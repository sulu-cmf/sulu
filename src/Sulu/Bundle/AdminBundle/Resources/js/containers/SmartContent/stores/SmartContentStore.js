// @flow
import {action, autorun, computed, observable, toJS} from 'mobx';
import type {IObservableValue} from 'mobx';
import Requester from '../../../services/Requester';
import Config from '../../../services/Config';
import ResourceRequester from '../../../services/ResourceRequester';
import {buildQueryString} from '../../../utils/Request';
import type {Conjunction, FilterCriteria, SortOrder} from '../types';

export default class SmartContentStore {
    provider: string;
    locale: ?IObservableValue<string>;
    dataSourceResourceKey: ?string;
    @observable items: Array<Object> = [];
    @observable itemsLoading: boolean;
    @observable categoriesLoading: boolean;
    @observable dataSourceLoading: boolean;
    @observable dataSource: ?Object;
    @observable includeSubElements: ?boolean;
    @observable categories: ?Array<Object>;
    @observable categoryOperator: ?Conjunction;
    @observable tags: ?Array<string | number>;
    @observable tagOperator: ?Conjunction;
    @observable audienceTargeting: ?boolean;
    @observable sortBy: ?string;
    @observable sortOrder: ?SortOrder;
    @observable presentation: ?string;
    @observable limit: ?number;
    itemDisposer: () => void;
    id: ?string | number;

    constructor(
        provider: string,
        filterCriteria: ?FilterCriteria,
        locale?: ?IObservableValue<string>,
        dataSourceResourceKey: ?string,
        id: ?string | number
    ) {
        this.provider = provider;
        this.locale = locale;
        this.dataSourceResourceKey = dataSourceResourceKey;
        this.id = id;

        if (filterCriteria) {
            this.audienceTargeting = filterCriteria.audienceTargeting;
            this.categoryOperator = filterCriteria.categoryOperator;
            this.includeSubElements = filterCriteria.includeSubFolders;
            this.limit = filterCriteria.limitResult;
            this.sortBy = filterCriteria.sortBy;
            this.sortOrder = filterCriteria.sortMethod;
            this.tagOperator = filterCriteria.tagOperator;
            this.tags = filterCriteria.tags;
            this.presentation = filterCriteria.presentAs;

            if (filterCriteria.categories) {
                this.categoriesLoading = true;
                // TODO extract 'categories' into some kind of variable?
                ResourceRequester.get(
                    'categories',
                    undefined,
                    {
                        ids: filterCriteria.categories,
                        locale: this.locale ? this.locale.get() : undefined,
                    }
                ).then(action((response) => {
                    this.categoriesLoading = false;
                    this.categories = response._embedded.categories;
                }));
            }

            if (filterCriteria.dataSource && this.dataSourceResourceKey) {
                this.dataSourceLoading = true;
                ResourceRequester.get(
                    this.dataSourceResourceKey,
                    filterCriteria.dataSource,
                    {locale: this.locale ? this.locale.get() : undefined}
                ).then(action((response) => {
                    this.dataSource = response;
                    this.dataSourceLoading = false;
                }));
            }
        }

        this.itemDisposer = autorun(this.loadItems);
    }

    destroy() {
        this.itemDisposer();
    }

    loadItems = () => {
        if (!this.hasFilterCriteria || this.loading) {
            this.setItems([]);
            return;
        }

        this.setItemsLoading(true);

        return Requester.get(
            Config.endpoints.items + buildQueryString({
                provider: this.provider,
                excluded: this.id,
                locale: this.locale,
                ...this.filterCriteria,
            })
        ).then(action((response) => {
            this.setItems(response._embedded.items);
            this.setItemsLoading(false);
        }));
    };

    @action setItems(items: Array<Object>) {
        this.items = items;
    }

    @action setItemsLoading(itemsLoading: boolean) {
        this.itemsLoading = itemsLoading;
    }

    @computed get loading() {
        return this.dataSourceLoading || this.categoriesLoading;
    }

    @computed get filterCriteria(): FilterCriteria {
        return {
            audienceTargeting: this.audienceTargeting,
            categories: this.categories && this.categories.length > 0
                ? this.categories.map((category) => category.id)
                : undefined,
            categoryOperator: this.categoryOperator,
            dataSource: this.dataSource ? this.dataSource.id : undefined,
            includeSubFolders: this.includeSubElements,
            limitResult: this.limit,
            sortBy: this.sortBy,
            sortMethod: this.sortOrder,
            tagOperator: this.tagOperator,
            tags: this.tags && this.tags.length > 0 ? toJS(this.tags) : undefined,
            presentAs: this.presentation,
        };
    }

    @computed get hasFilterCriteria(): boolean {
        const {dataSource, categories, tags} = this.filterCriteria;

        return dataSource !== undefined
            || (!!categories && categories.length > 0)
            || (!!tags && tags.length > 0);
    }
}
