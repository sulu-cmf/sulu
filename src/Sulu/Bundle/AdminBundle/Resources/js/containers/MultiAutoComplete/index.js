// @flow
import MultiAutoComplete from './MultiAutoComplete';

export default MultiAutoComplete;
