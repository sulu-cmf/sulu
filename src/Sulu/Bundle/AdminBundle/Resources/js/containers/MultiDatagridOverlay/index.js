// @flow
import MultiDatagridOverlay from './MultiDatagridOverlay';

export default MultiDatagridOverlay;
