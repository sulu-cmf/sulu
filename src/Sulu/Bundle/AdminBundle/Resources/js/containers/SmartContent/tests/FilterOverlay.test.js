// @flow
import React from 'react';
import {shallow, mount} from 'enzyme';
import SmartContentStore from '../stores/SmartContentStore';
import MultiAutoComplete from '../../../containers/MultiAutoComplete';
import MultiDatagridOverlay from '../../../containers/MultiDatagridOverlay';
import SingleDatagridOverlay from '../../../containers/SingleDatagridOverlay';
import FilterOverlay from '../FilterOverlay';

jest.mock('../stores/SmartContentStore', () => jest.fn());

jest.mock('../../../utils/Translator', () => ({
    translate: jest.fn((key) => key),
}));

jest.mock('../../../containers/MultiAutoComplete', () => jest.fn(() => null));
jest.mock('../../../containers/MultiDatagridOverlay', () => jest.fn(() => null));
jest.mock('../../../containers/SingleDatagridOverlay', () => jest.fn(() => null));

test('Do not display if open is set to false', () => {
    const smartContentStore = new SmartContentStore('content');
    const filterOverlay = shallow(
        <FilterOverlay
            dataSourceAdapter="table"
            dataSourceResourceKey="snippets"
            onClose={jest.fn()}
            open={false}
            presentations={{}}
            sections={[]}
            smartContentStore={smartContentStore}
            sortings={{}}
            title="Test"
        />
    );

    expect(filterOverlay.find('Overlay').prop('open')).toEqual(false);
});

test('Render with DatagridOverlays if smartContentStore is loaded', () => {
    const smartContentStore = new SmartContentStore('content');
    // $FlowFixMe
    smartContentStore.loading = false;

    const filterOverlay = shallow(
        <FilterOverlay
            dataSourceAdapter="table"
            dataSourceResourceKey="snippets"
            onClose={jest.fn()}
            open={true}
            presentations={{}}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{}}
            title="Test"
        />
    );

    expect(filterOverlay.find(SingleDatagridOverlay)).toHaveLength(1);
    expect(filterOverlay.find(MultiDatagridOverlay)).toHaveLength(1);
});

test('Render without DatagridOverlays if smartContentStore is not loaded', () => {
    const smartContentStore = new SmartContentStore('content');
    // $FlowFixMe
    smartContentStore.loading = true;

    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter={undefined}
            dataSourceResourceKey={undefined}
            onClose={jest.fn()}
            open={true}
            presentations={{}}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{}}
            title="Test"
        />
    );
    expect(filterOverlay.find(SingleDatagridOverlay)).toHaveLength(0);
    expect(filterOverlay.find(MultiDatagridOverlay)).toHaveLength(0);
});

test('Render with all fields', () => {
    const smartContentStore = new SmartContentStore('content');
    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter={undefined}
            dataSourceResourceKey={undefined}
            onClose={jest.fn()}
            open={true}
            presentations={{}}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{}}
            title="Test"
        />
    );
    expect(filterOverlay.find('Portal').at(2).render()).toMatchSnapshot();
});

test('Render with no fields', () => {
    const smartContentStore = new SmartContentStore('content');
    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter={undefined}
            dataSourceResourceKey={undefined}
            onClose={jest.fn()}
            open={true}
            presentations={{}}
            sections={[]}
            smartContentStore={smartContentStore}
            sortings={{}}
            title="Test"
        />
    );
    expect(filterOverlay.find('Portal').at(2).render()).toMatchSnapshot();
});

test('Fill all fields using and update SmartContentStore on confirm', () => {
    const smartContentStore = new SmartContentStore('content');
    const closeSpy = jest.fn();

    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter="table"
            dataSourceResourceKey="pages"
            onClose={closeSpy}
            open={true}
            presentations={{
                small: 'Small',
                large: 'Large',
            }}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{
                title: 'Title',
                changed: 'Changed',
            }}
            title="Test"
        />
    );

    filterOverlay.find('Button[children="sulu_admin.choose_data_source"]').prop('onClick')();
    filterOverlay.update();
    expect(filterOverlay.find(SingleDatagridOverlay).find({resourceKey: 'pages'}).prop('open')).toEqual(true);
    filterOverlay.find(SingleDatagridOverlay).find({resourceKey: 'pages'}).prop('onConfirm')({id: 2, title: 'Test'});
    filterOverlay.update();
    expect(filterOverlay.find(SingleDatagridOverlay).find({resourceKey: 'pages'}).prop('open')).toEqual(false);
    expect(filterOverlay.find('section').at(1).find('label[className="description"]').text())
        .toEqual('sulu_admin.data_source: Test');

    filterOverlay.find('Toggler[children="sulu_admin.include_sub_elements"]').prop('onChange')(true);
    filterOverlay.update();
    expect(filterOverlay.find('Toggler[children="sulu_admin.include_sub_elements"]').prop('checked')).toEqual(true);

    filterOverlay.find('Button[children="sulu_admin.choose_categories"]').prop('onClick')();
    filterOverlay.update();
    expect(filterOverlay.find(MultiDatagridOverlay).find({resourceKey: 'categories'}).prop('open')).toEqual(true);
    filterOverlay.find(MultiDatagridOverlay).find({resourceKey: 'categories'}).prop('onConfirm')([
        {id: 1, name: 'Test1'},
        {id: 3, name: 'Test2'},
    ]);
    filterOverlay.update();
    expect(filterOverlay.find(MultiDatagridOverlay).find({resourceKey: 'categories'}).prop('open')).toEqual(false);
    expect(filterOverlay.find('section').at(2).find('label[className="description"]').text())
        .toEqual('sulu_category.categories: Test1, Test2');

    filterOverlay.find('div[className="categories"]').find('SingleSelect').prop('onChange')('and');
    filterOverlay.update();
    expect(filterOverlay.find('div[className="categories"]').find('SingleSelect').prop('value')).toEqual('and');

    filterOverlay.find(MultiAutoComplete).prop('onChange')(['Test1', 'Test3']);
    filterOverlay.update();
    expect(filterOverlay.find(MultiAutoComplete).prop('value')).toEqual(['Test1', 'Test3']);

    filterOverlay.find('div[className="tags"]').find('SingleSelect').prop('onChange')('or');
    filterOverlay.update();
    expect(filterOverlay.find('div[className="tags"]').find('SingleSelect').prop('value')).toEqual('or');

    filterOverlay.find('Toggler[children="sulu_admin.use_target_groups"]').prop('onChange')(false);
    filterOverlay.update();
    expect(filterOverlay.find('Toggler[children="sulu_admin.use_target_groups"]').prop('checked')).toEqual(false);

    filterOverlay.find('div[className="sortColumn"]').find('SingleSelect').prop('onChange')('changed');
    filterOverlay.update();
    expect(filterOverlay.find('div[className="sortColumn"]').find('SingleSelect').prop('value')).toEqual('changed');

    filterOverlay.find('div[className="sortOrder"]').find('SingleSelect').prop('onChange')('asc');
    filterOverlay.update();
    expect(filterOverlay.find('div[className="sortOrder"]').find('SingleSelect').prop('value')).toEqual('asc');

    filterOverlay.find('div[className="presentation"]').find('SingleSelect').prop('onChange')('large');
    filterOverlay.update();
    expect(filterOverlay.find('div[className="presentation"]').find('SingleSelect').prop('value')).toEqual('large');

    filterOverlay.find('div[className="limit"] Number').prop('onChange')(7);
    filterOverlay.update();
    expect(filterOverlay.find('div[className="limit"] Number').prop('value')).toEqual(7);

    expect(smartContentStore.dataSource).toEqual(undefined);
    expect(smartContentStore.includeSubElements).toEqual(undefined);
    expect(smartContentStore.categories).toEqual(undefined);
    expect(smartContentStore.categoryOperator).toEqual(undefined);
    expect(smartContentStore.tags).toEqual(undefined);
    expect(smartContentStore.tagOperator).toEqual(undefined);
    expect(smartContentStore.audienceTargeting).toEqual(undefined);
    expect(smartContentStore.sortBy).toEqual(undefined);
    expect(smartContentStore.sortOrder).toEqual(undefined);
    expect(smartContentStore.presentation).toEqual(undefined);
    expect(smartContentStore.limit).toEqual(undefined);

    filterOverlay.find('Overlay').prop('onConfirm')();

    expect(smartContentStore.dataSource).toEqual({id: 2, title: 'Test'});
    expect(smartContentStore.includeSubElements).toEqual(true);
    expect(smartContentStore.categories).toEqual([{id: 1, name: 'Test1'}, {id: 3, name: 'Test2'}]);
    expect(smartContentStore.categoryOperator).toEqual('and');
    expect(smartContentStore.tags).toEqual(['Test1', 'Test3']);
    expect(smartContentStore.tagOperator).toEqual('or');
    expect(smartContentStore.audienceTargeting).toEqual(false);
    expect(smartContentStore.sortBy).toEqual('changed');
    expect(smartContentStore.sortOrder).toEqual('asc');
    expect(smartContentStore.presentation).toEqual('large');
    expect(smartContentStore.limit).toEqual(7);

    expect(closeSpy).toBeCalledWith();
});

test('Prefill all fields with correct values', () => {
    const smartContentStore = new SmartContentStore('content');
    smartContentStore.dataSource = {id: 4, title: 'Homepage'};
    smartContentStore.includeSubElements = true;
    smartContentStore.categories = [{id: 1, name: 'Test1'}, {id: 5, name: 'Test3'}];
    smartContentStore.categoryOperator = 'or';
    smartContentStore.tags = ['Test5', 'Test7'];
    smartContentStore.tagOperator = 'and';
    smartContentStore.audienceTargeting = true;
    smartContentStore.sortBy = 'created';
    smartContentStore.sortOrder = 'desc';
    smartContentStore.presentation = 'small';
    smartContentStore.limit = 8;

    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter="table"
            dataSourceResourceKey="pages"
            onClose={jest.fn()}
            open={true}
            presentations={{
                small: 'Small',
                large: 'Large',
            }}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{
                title: 'Title',
                created: 'Created',
            }}
            title="Test"
        />
    );

    expect(filterOverlay.find('section').at(1).find('label[className="description"]').text())
        .toEqual('sulu_admin.data_source: Homepage');
    expect(filterOverlay.find(SingleDatagridOverlay).find({resourceKey: 'pages'}).prop('preSelectedItem'))
        .toEqual({id: 4, title: 'Homepage'});
    expect(filterOverlay.find('Toggler[children="sulu_admin.include_sub_elements"]').prop('checked')).toEqual(true);

    expect(filterOverlay.find('section').at(2).find('label[className="description"]').text())
        .toEqual('sulu_category.categories: Test1, Test3');
    expect(filterOverlay.find(MultiDatagridOverlay).find({resourceKey: 'categories'}).prop('preSelectedItems'))
        .toEqual([{id: 1, name: 'Test1'}, {id: 5, name: 'Test3'}]);
    expect(filterOverlay.find('div[className="categories"]').find('SingleSelect').prop('value')).toEqual('or');

    expect(filterOverlay.find(MultiAutoComplete).prop('value')).toEqual(['Test5', 'Test7']);
    expect(filterOverlay.find('div[className="tags"]').find('SingleSelect').prop('value')).toEqual('and');

    expect(filterOverlay.find('Toggler[children="sulu_admin.use_target_groups"]').prop('checked')).toEqual(true);

    expect(filterOverlay.find('div[className="sortColumn"]').find('SingleSelect').prop('value')).toEqual('created');
    expect(filterOverlay.find('div[className="sortOrder"]').find('SingleSelect').prop('value')).toEqual('desc');

    expect(filterOverlay.find('div[className="presentation"]').find('SingleSelect').prop('value')).toEqual('small');
    expect(filterOverlay.find('div[className="limit"] Number').prop('value')).toEqual(8);
});

test('Reset all fields when reset action is clicked', () => {
    const smartContentStore = new SmartContentStore('content');
    smartContentStore.dataSource = {id: 4, url: '/home'};
    smartContentStore.includeSubElements = true;
    smartContentStore.categories = [{id: 1, name: 'Test1'}, {id: 5, name: 'Test3'}];
    smartContentStore.categoryOperator = 'or';
    smartContentStore.tags = ['Test5', 'Test7'];
    smartContentStore.tagOperator = 'and';
    smartContentStore.audienceTargeting = true;
    smartContentStore.sortBy = 'created';
    smartContentStore.sortOrder = 'desc';
    smartContentStore.presentation = 'large';
    smartContentStore.limit = 5;

    const filterOverlay = mount(
        <FilterOverlay
            dataSourceAdapter="table"
            dataSourceResourceKey="pages"
            onClose={jest.fn()}
            open={true}
            presentations={{
                small: 'Small',
                large: 'Large',
            }}
            sections={['datasource', 'categories', 'tags', 'audienceTargeting', 'sorting', 'presentation', 'limit']}
            smartContentStore={smartContentStore}
            sortings={{
                title: 'Title',
                created: 'Created',
            }}
            title="Test"
        />
    );

    filterOverlay.find('Overlay').prop('actions')[0].onClick();
    filterOverlay.update();

    expect(filterOverlay.instance().dataSource).toEqual(undefined);
    expect(filterOverlay.instance().includeSubElements).toEqual(undefined);
    expect(filterOverlay.instance().categories).toEqual(undefined);
    expect(filterOverlay.instance().categoryOperator).toEqual(undefined);
    expect(filterOverlay.instance().tags).toEqual(undefined);
    expect(filterOverlay.instance().tagOperator).toEqual(undefined);
    expect(filterOverlay.instance().audienceTargeting).toEqual(undefined);
    expect(filterOverlay.instance().sortBy).toEqual(undefined);
    expect(filterOverlay.instance().sortOrder).toEqual(undefined);
    expect(filterOverlay.instance().presentation).toEqual(undefined);
    expect(filterOverlay.instance().limit).toEqual(undefined);
});
