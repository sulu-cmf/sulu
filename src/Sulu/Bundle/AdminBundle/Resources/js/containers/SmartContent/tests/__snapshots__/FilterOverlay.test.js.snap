// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render with all fields 1`] = `
<div
  class="container"
>
  <div
    class="overlay small"
  >
    <section
      class="content"
    >
      <header>
        <h2>
          Test
        </h2>
        <span
          aria-label="su-times"
          class="su-times clickable icon"
          role="button"
          tabindex="0"
        />
      </header>
      <article>
        <div
          class="content"
        >
          <section
            class="section"
          >
            <h3>
              sulu_admin.data_source
            </h3>
            <div
              class="source"
            >
              <button
                class="button secondary sourceButton"
                type="button"
              >
                <span
                  class="text"
                >
                  sulu_admin.choose_data_source
                </span>
              </button>
              <label
                class="label"
              >
                <span
                  class="switch toggler"
                >
                  <input
                    type="checkbox"
                    value=""
                  />
                  <span />
                </span>
                <div>
                  sulu_admin.include_sub_elements
                </div>
              </label>
            </div>
            <label
              class="description"
            >
              sulu_admin.data_source: 
            </label>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.filter_by_categories
            </h3>
            <div
              class="categories"
            >
              <button
                class="button secondary"
                type="button"
              >
                <span
                  class="text"
                >
                  sulu_admin.choose_categories
                </span>
              </button>
              <div
                class="categoriesSelect"
              >
                <div
                  class="select"
                >
                  <button
                    class="displayValue default"
                    type="button"
                  >
                    <div
                      aria-label="sulu_admin.any_category_description"
                      class="croppedText"
                      title="sulu_admin.any_category_description"
                    >
                      <div
                        aria-hidden="true"
                        class="front"
                      >
                        sulu_admin.any_cat
                      </div>
                      <div
                        aria-hidden="true"
                        class="back"
                      >
                        <span>
                          egory_description
                        </span>
                      </div>
                      <div
                        class="whole"
                      >
                        sulu_admin.any_category_description
                      </div>
                    </div>
                    <span
                      aria-label="su-angle-down"
                      class="su-angle-down toggle"
                    />
                  </button>
                </div>
              </div>
            </div>
            <label
              class="description"
            >
              sulu_category.categories: 
            </label>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.filter_by_tags
            </h3>
            <div
              class="tags"
            >
              <div
                class="tagsAutoComplete"
              />
              <div
                class="tagsSelect"
              >
                <div
                  class="select"
                >
                  <button
                    class="displayValue default"
                    type="button"
                  >
                    <div
                      aria-label="sulu_admin.any_tag_description"
                      class="croppedText"
                      title="sulu_admin.any_tag_description"
                    >
                      <div
                        aria-hidden="true"
                        class="front"
                      >
                        sulu_admin.any_
                      </div>
                      <div
                        aria-hidden="true"
                        class="back"
                      >
                        <span>
                          tag_description
                        </span>
                      </div>
                      <div
                        class="whole"
                      >
                        sulu_admin.any_tag_description
                      </div>
                    </div>
                    <span
                      aria-label="su-angle-down"
                      class="su-angle-down toggle"
                    />
                  </button>
                </div>
              </div>
            </div>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.target_groups
            </h3>
            <label
              class="label"
            >
              <span
                class="switch toggler"
              >
                <input
                  type="checkbox"
                  value=""
                />
                <span />
              </span>
              <div>
                sulu_admin.use_target_groups
              </div>
            </label>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.sort_by
            </h3>
            <div
              class="sorting"
            >
              <div
                class="sortColumn"
              >
                <div
                  class="select"
                >
                  <button
                    class="displayValue default"
                    type="button"
                  >
                    <span
                      aria-label="su-angle-down"
                      class="su-angle-down toggle"
                    />
                  </button>
                </div>
              </div>
              <div
                class="sortOrder"
              >
                <div
                  class="select"
                >
                  <button
                    class="displayValue default"
                    type="button"
                  >
                    <div
                      aria-label="sulu_admin.ascending"
                      class="croppedText"
                      title="sulu_admin.ascending"
                    >
                      <div
                        aria-hidden="true"
                        class="front"
                      >
                        sulu_admin
                      </div>
                      <div
                        aria-hidden="true"
                        class="back"
                      >
                        <span>
                          .ascending
                        </span>
                      </div>
                      <div
                        class="whole"
                      >
                        sulu_admin.ascending
                      </div>
                    </div>
                    <span
                      aria-label="su-angle-down"
                      class="su-angle-down toggle"
                    />
                  </button>
                </div>
              </div>
            </div>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.present_as
            </h3>
            <div
              class="presentation"
            >
              <div
                class="select"
              >
                <button
                  class="displayValue default"
                  type="button"
                >
                  <span
                    aria-label="su-angle-down"
                    class="su-angle-down toggle"
                  />
                </button>
              </div>
            </div>
          </section>
          <section
            class="section"
          >
            <h3>
              sulu_admin.limit_result_to
            </h3>
            <div
              class="limit"
            >
              <label
                class="input default left"
              >
                <input
                  type="number"
                  value=""
                />
              </label>
            </div>
          </section>
        </div>
      </article>
      <footer>
        <div
          class="actions"
        >
          <button
            class="button link"
            type="button"
          >
            <span
              class="text"
            >
              sulu_admin.reset
            </span>
          </button>
        </div>
        <button
          class="button primary"
          type="button"
        >
          <span
            class="text"
          >
            sulu_admin.confirm
          </span>
        </button>
      </footer>
    </section>
  </div>
</div>
`;

exports[`Render with no fields 1`] = `
<div
  class="container"
>
  <div
    class="overlay small"
  >
    <section
      class="content"
    >
      <header>
        <h2>
          Test
        </h2>
        <span
          aria-label="su-times"
          class="su-times clickable icon"
          role="button"
          tabindex="0"
        />
      </header>
      <article>
        <div
          class="content"
        />
      </article>
      <footer>
        <div
          class="actions"
        >
          <button
            class="button link"
            type="button"
          >
            <span
              class="text"
            >
              sulu_admin.reset
            </span>
          </button>
        </div>
        <button
          class="button primary"
          type="button"
        >
          <span
            class="text"
          >
            sulu_admin.confirm
          </span>
        </button>
      </footer>
    </section>
  </div>
</div>
`;
